### PR TITLE
ICS28: CCV draft w/ complete unbonding

### PR DESCRIPTION
**Pull request by mpoke**
The partial specification of Cross-Chain Validation. 

Changes to base (i.e., [ccv](https://github.com/cosmos/ibc/tree/marius/ccv-init-and-vsc)):
- The staking hooks that enable the timely completion of unbonding operations (e.g., undelegations) on the provider chain (i.e., waiting for the operations to mature on the consumer chain).

Features not included (to be added to other PRs):
- the evidence / slashing subprotocol that enables the provider chain to punish validators for misbehaviors on the consumer chain;
- the distribution subprotocol that enables the validators to receive rewards for validating on the consumer chain.

## Prerequisites 

- [x] Handle issue https://github.com/cosmos/interchain-security/issues/33 before merging

### CODE REVIEW

File: spec/app/ics-028-cross-chain-validation/technical_specification.md
Comment (josef-widder) [inline spec/app/ics-028-cross-chain-validation/technical_specification.md:1027]: I wonder whether the pseudo code provides additional information that is not present in the expected postconditions? In general, is there is no algorithmic ideas, I have tried to reduce pseudo code as much as possible from specs. But perhaps that is against interchain standards?

**Review by josef-widder — APPROVED** [review]
Looks good to me. The only question is whether pseudo code and the bullet points below the functions may not be redundant. We can discuss this.

**Review by AdityaSripal — APPROVED** [review]
Nice work!

**Comment by AdityaSripal** [comment]
I would hold off on merging until the init spec is approved so it is easier to review

**Comment by mpoke** [comment]
- [ ] Handle issue https://github.com/cosmos/interchain-security/issues/33 before merging

File: spec/app/ics-028-cross-chain-validation/overview_and_basic_concepts.md
Comment (josef-widder) [inline spec/app/ics-028-cross-chain-validation/overview_and_basic_concepts.md:179]:
```suggestion
In the context of CCV, the completion MUST require also the unbonding operation to reach maturity on **all** consumer chains (for the [Security Model](#security-model) to be preserved). 
```

**Review by josef-widder — APPROVED** [review]

File: spec/app/ics-028-cross-chain-validation/overview_and_basic_concepts.md
Comment (mpoke) [reply spec/app/ics-028-cross-chain-validation/overview_and_basic_concepts.md:102]: It's from the SDK documentation, see https://docs.cosmos.network/v0.44/modules/staking/05_end_block.html#unbonding-validators

**Pull request merged**
